CHANGELOG/v1.10.0-beta.1.md

```diff
@@ -0,0 +1,449 @@
+🚨 This is a BETA RELEASE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/new).
+
+:warning: **BETA RELEASE NOTES** :warning:
+
+## Highlights
+- Bumped to controller-runtime v0.20, k8s.io/* v0.32, controller-gen v0.17
+- Features:
+  - ClusterResourceSet was promoted to GA (#11365)
+  - PriorityQueue was added as an alpha feature (see kubernetes-sigs/controller-runtime#2374 for more details) (#11698)
+- API: Added additional validation to our CRDs (#11834)
+- API: Various improvements to v1beta2 conditions
+- ClusterClass: Support referencing ClusterClasses across namespaces (#11352 #11395 #11649)
+- ClusterClass: Add NamingStrategy for InfraCluster (#11898)
+- Machine: Add MachineDrainRule behavior "WaitCompleted" (#11545)
+- Machine: Add --additional-sync-machine-labels flag to allow syncing additional labels to Nodes (#11650)
+- Machine: Sync Machine annotations to Nodes (#11813)
+- MachineSet/MachineDeployment: Add NamingStrategy to MachineDeployment (#11172)
+- MachineSet/KCP: Improve preflight checks (#11927 #11941 #11928)
+- CRD migration: Add CRD migrator (#11889)
+- Metrics: Add ClusterCache (#11789) and SSA cache (#11635) metrics
+- Merged CAPIM in CAPD (see From CAPD(docker) to CAPD(dev) for more details)
+- As usual, significantly improved e2e test framework & coverage (e.g. #11667 #11884 #11966 #9620 #11983)
+- New providers in clusterctl:
+  - Addon providers: fleet (#11806)
+  - Infrastructure providers: OpenNebula (#11835), Harvester (#11477), Huawei Cloud (#11861)
+  
+## Deprecation and Removals Warning
+- KCP/MachineSet: Removed deprecated flag for old infra machine naming (#11679)
+- MD/MS/MP/KCP: Deprecated status replica counters planned for removal (#11516)
+- MachineDeployment: Deprecated spec.progressDeadlineSeconds (#11472)
+- CRD migration: Deprecated clusterctl upgrade CRD storage version migration for providers (#11889)
+- ClusterResourceSet was promoted to GA, feature flag is now deprecated (#11741)
+
+## Changes since v1.10.0-beta.0
+## :chart_with_upwards_trend: Overview
+- 14 new commits merged
+- 1 feature addition ✨
+- 6 bugs fixed 🐛
+
+## :sparkles: New Features
+- Runtime SDK/ClusterClass: Extend Cluster builtin to include metadata (#12014)
+
+## :bug: Bug Fixes
+- CABPK: Make KubeadmConfig FileSystem.Label optional (#12019)
+- CAPD: Fix worker machine count in CAPD template (#12028)
+- CAPIM: Fix CAPD in-memory templates (#12013)
+- clusterctl: Clusterctl upgrade hangs for a time on CRD migration when new version contains a number of new CRDs (#11984)
+- e2e: Stop overwriting ExtraPortMappings if WithDockerSockMount option is used (#12012)
+- util: CRD migration: Fix cases where update validation fails (#11991)
```

Comment:
This was not cherry-picked onto release-1.10. I think we have a problem in release notes generation

(please also check all other entries here + if there are any missing)

Reply:
One moment

Reply:
Add this comment to the wrong line 12017 is not on the release branch (l.53 below)

Reply:
Yes, there must be some bug, probably because the release-notes generator is still looking at `main`? We will fix.

I went through the actual diff https://github.com/kubernetes-sigs/cluster-api/compare/v1.10.0-beta.0...release-1.10 and compared and updated this section, so I think it's accurate now.